cmd/scollector/collectors/processes_linux.go

```diff
@@ -25,6 +25,23 @@ func AddProcessConfig(params conf.ProcessParams) error {
 
 var watchedProcs = []*WatchedProc{}
 
+func linuxCoreCount() (c int64, err error) {
+	if err = readLine("/proc/cpuinfo", func(s string) (err error) {
+		f := strings.Fields(s)
+		if len(f) > 2 && f[0] == "processor" {
```

Comment:
This will lead to an erroneous core count when hyperthreading is in use, as logical cores get their own processor entry in /proc/cpuinfo.

For example, my personal desktop has 8 processors listed in /proc/cpuinfo, but it is only a 4-core system. You can tell as the 'core id' is duplicated for 4 of em.

I *believe* you can instead count the 'core id' field to get this.